// @crayonai/react-core/dist/hooks/useThreadState.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useThreadState = void 0;
const zustand_1 = require("zustand");
const ChatContext_1 = require("../internal/ChatContext");
/**
 * `useThreadState` allows you to access the {@link ThreadState}. This is helpful for multiple reasons, including but not limited to:
 *
 * - You can use the state to render the thread UI
 * - You can use the state to trigger actions on the thread
 *
 * @category Hooks
 */
const useThreadState = () => {
    const { threadManager } = (0, ChatContext_1.useChatContext)();
    return {
        isLoadingMessages: (0, zustand_1.useStore)(threadManager, (store) => store.isLoadingMessages),
        isRunning: (0, zustand_1.useStore)(threadManager, (store) => store.isRunning),
        messages: (0, zustand_1.useStore)(threadManager, (store) => store.messages),
        error: (0, zustand_1.useStore)(threadManager, (store) => store.error),
        responseTemplates: (0, zustand_1.useStore)(threadManager, (store) => store.responseTemplates),
    };
};
exports.useThreadState = useThreadState;
