// @crayonai/react-core/dist/hooks/useThreadListActions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useThreadListActions = void 0;
const zustand_1 = require("zustand");
const ChatContext_1 = require("../internal/ChatContext");
/**
 * `useThreadListActions` allows you to modify the {@link ThreadListState} by providing access to {@link ThreadListActions} which contains methods to modify the state.
 *
 * @category Hooks
 */
const useThreadListActions = () => {
    const { threadListManager } = (0, ChatContext_1.useChatContext)();
    return {
        load: (0, zustand_1.useStore)(threadListManager, (store) => store.load),
        switchToNewThread: (0, zustand_1.useStore)(threadListManager, (store) => store.switchToNewThread),
        selectThread: (0, zustand_1.useStore)(threadListManager, (store) => store.selectThread),
        createThread: (0, zustand_1.useStore)(threadListManager, (store) => store.createThread),
        deleteThread: (0, zustand_1.useStore)(threadListManager, (store) => store.deleteThread),
        updateThread: (0, zustand_1.useStore)(threadListManager, (store) => store.updateThread),
    };
};
exports.useThreadListActions = useThreadListActions;
