// @crayonai/react-core/dist/hooks/useThreadListState.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useThreadListState = void 0;
const zustand_1 = require("zustand");
const ChatContext_1 = require("../internal/ChatContext");
/**
 * `useThreadListState` allows you to access the {@link ThreadListState}. This is helpful for multiple reasons, including but not limited to:
 *
 * - You can use the state to render the thread list UI
 * - You can use the state to trigger actions on the thread list
 *
 * @category Hooks
 */
const useThreadListState = () => {
    const { threadListManager } = (0, ChatContext_1.useChatContext)();
    return {
        isLoading: (0, zustand_1.useStore)(threadListManager, (store) => store.isLoading),
        threads: (0, zustand_1.useStore)(threadListManager, (store) => store.threads),
        error: (0, zustand_1.useStore)(threadListManager, (store) => store.error),
        selectedThreadId: (0, zustand_1.useStore)(threadListManager, (store) => store.selectedThreadId),
        shouldResetThreadState: (0, zustand_1.useStore)(threadListManager, (store) => store.shouldResetThreadState),
    };
};
exports.useThreadListState = useThreadListState;
