// @crayonai/react-core/dist/hooks/useThreadActions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useThreadActions = void 0;
const zustand_1 = require("zustand");
const ChatContext_1 = require("../internal/ChatContext");
/**
 * `useThreadActions` allows you to modify the {@link ThreadState} by providing access to {@link ThreadActions} which contains methods to modify the state.
 *
 * @category Hooks
 */
const useThreadActions = () => {
    const { threadManager } = (0, ChatContext_1.useChatContext)();
    return {
        processMessage: (0, zustand_1.useStore)(threadManager, (store) => store.processMessage),
        appendMessages: (0, zustand_1.useStore)(threadManager, (store) => store.appendMessages),
        updateMessage: (0, zustand_1.useStore)(threadManager, (store) => store.updateMessage),
        setMessages: (0, zustand_1.useStore)(threadManager, (store) => store.setMessages),
        onCancel: (0, zustand_1.useStore)(threadManager, (store) => store.onCancel),
        deleteMessage: (0, zustand_1.useStore)(threadManager, (store) => store.deleteMessage),
    };
};
exports.useThreadActions = useThreadActions;
