// @thesysai/genui-sdk/dist/xmlParser-URxwKJGT.js
import { Parser as T } from "htmlparser2";
import { produce as u } from "immer";
import m from "lodash/escape";
import $ from "lodash/findLastIndex";
import x from "lodash/unescape";
var C = ((t) => (t.PRESENTATION = "presentation", t.SLIDES = "slides", t.REPORT = "report", t))(C || {}), b = ((t) => (t.CONTENT = "content", t.CUSTOM_MARKDOWN = "custommarkdown", t.ARTIFACT = "artifact", t))(b || {});
const r = (t, e) => `<${t}${e ? ` ${Object.entries(e).map(([n, s]) => `${n}="${s}"`).join(" ")}` : ""}>`, f = (t) => `</${t}>`;
function w(t, e) {
  switch (e.currentTag) {
    case "content":
      e.messages[h(e)].data += t;
      break;
    case "artifact":
      e.messages[k(e)].data += t;
      break;
    case "artifact_diff": {
      const n = k(e);
      n !== -1 && (e.messages[n].diff += t);
      break;
    }
    case "context":
      e.context += t;
      break;
    case "thinkitem":
      break;
    case "thinkitemtitle": {
      const n = e.think[e.think.length - 1];
      n && (e.think = e.think.slice(0, -1).concat({ ...n, title: n.title + t }));
      break;
    }
    case "thinkitemcontent": {
      const n = e.think[e.think.length - 1];
      n && (e.think = e.think.slice(0, -1).concat({ ...n, content: n.content + t }));
      break;
    }
    case "custommarkdown": {
      const n = e.messages[(function(i) {
        return $(i.messages, (a) => a.type === "custommarkdown");
      })(e)], s = n.data;
      s && (n.data = { ...s, content: s.content + t });
      break;
    }
    default: {
      const n = h(e);
      e.messages[n] ? e.messages[n].data += t : e.messages.push({ type: "content", data: t });
    }
  }
}
function k(t) {
  return t.messages.findIndex((e) => e.type === "artifact");
}
function h(t) {
  return t.messages.findIndex((e) => e.type === "content");
}
function y() {
  let t = { isContentTagClosed: !1, messages: [], think: [], context: "", currentTag: void 0 };
  const e = new T({ onopentag: (n, s) => {
    t = u(t, (i) => {
      (function(a, g, c) {
        switch (a.currentTag = g, a.currentTag) {
          case "content": {
            const o = a.messages.find((d) => d.type === "content");
            a.isContentTagClosed = !1, o ? a.messages[h(a)].data = "" : a.messages.push({ type: "content", data: "" });
            break;
          }
          case "artifact": {
            const o = c.id ?? crypto.randomUUID(), d = c.version ? parseInt(c.version) : 1, l = a.messages.findIndex((p) => p.type === "artifact" && p.id === o);
            l === -1 ? a.messages.push({ type: "artifact", artifactType: c.type ? c.type : "slides", data: "", diff: "", version: d, id: o, isArtifactDiffTagClosed: !0 }) : a.messages[l].data = "";
            break;
          }
          case "artifact_diff": {
            const o = a.messages.find((d) => d.type === "artifact");
            o ? (o.diff = "", o.isArtifactDiffTagClosed = !1) : console.error("Artifact diff tag found without artifact tag");
            break;
          }
          case "context":
            a.context = "";
            break;
          case "thinkitem":
            a.think = a.think.concat({ title: "", content: "", ephemeral: c.ephemeral === "true" });
            break;
          case "custommarkdown":
            a.messages.push({ type: "custommarkdown", data: { content: "" } });
        }
      })(i, n, s);
    });
  }, ontext: (n) => {
    t = u(t, (s) => {
      (function(i, a) {
        w(a, i);
      })(s, n);
    });
  }, onclosetag: () => {
    t = u(t, (n) => {
      (function(s) {
        if (s.currentTag !== "content" && s.currentTag !== "artifact" && s.currentTag !== "artifact_diff" || (s.isContentTagClosed = !0), s.currentTag === "artifact_diff") {
          const i = s.messages[k(s)];
          i && (i.isArtifactDiffTagClosed = !0);
        }
        s.currentTag = void 0;
      })(n);
    });
  } }, { xmlMode: !0, decodeEntities: !0 });
  return { write: (n) => e.write(n), end: () => ({ messageParts: t.messages, think: t.think, context: t.context, isContentTagClosed: t.isContentTagClosed }), getResult: () => ({ isContentTagClosed: t.isContentTagClosed, messageParts: t.messages, think: t.think, context: t.context }), reset: () => {
    t = { isContentTagClosed: !1, messages: [], think: [], context: "", currentTag: void 0 };
  } };
}
function D(t) {
  const e = y();
  return e.write(t), e.end();
}
function E(t, e) {
  const n = m(t);
  return `${r("content")}${n}${f("content")}`;
}
function N(t) {
  return `${r("context")}${m(t)}${f("context")}`;
}
function P(t, e, n) {
  const s = `${r("thinkitemtitle")}${m(t)}${f("thinkitemtitle")}`, i = `${r("thinkitemcontent")}${m(e)}${f("thinkitemcontent")}`;
  return `${r("thinkitem", n ? { ephemeral: "true" } : void 0)}${s}${i}${f("thinkitem")}`;
}
function _(t) {
  const e = "custommarkdown";
  return `${r(e)}${m(t)}${f(e)}`;
}
const M = (t) => {
  const e = "<context>", n = "</context>", s = t.indexOf(e), i = t.indexOf(n);
  if (s !== -1 && i > s) {
    const a = t.slice(0, s) + t.slice(i + 10, t.length), g = t.slice(s + 9, i);
    return { c1Response: a, context: x(g) };
  }
  return { c1Response: t, context: "" };
};
export {
  C as A,
  b as M,
  N as a,
  _ as b,
  y as c,
  P as d,
  D as p,
  M as s,
  E as w
};
