// @crayonai/react-ui/dist/components/Tabs/stories/tabs.stories.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { BusFront, PlaneTakeoff, Ship } from "lucide-react";
import { Card } from "../../Card/Card";
import { CardHeader } from "../../CardHeader";
import { IconButton } from "../../IconButton";
import { Image } from "../../Image";
import { Tabs, TabsContent, TabsList, TabsTrigger } from "../Tabs";
const meta = {
    title: "Components/Tabs",
    component: Tabs,
    tags: ["!dev", "autodocs"],
    parameters: {
        layout: "centered",
        docs: {
            description: {
                component: "```tsx\nimport { Tabs, TabsContent, TabsList, TabsTrigger } from '@crayon-ui/react-ui';\n```",
            },
        },
    },
    decorators: [
        (Story) => (_jsx("div", { style: { width: "400px", margin: "2rem" }, children: _jsx(Story, {}) })),
    ],
    argTypes: {
        variant: {
            control: {
                type: "select",
                options: ["clear", "card", "sunk"],
            },
            description: "The variant of the tabs",
            table: {
                type: {
                    summary: "clear | card | sunk",
                },
                defaultValue: { summary: "clear" },
                category: "Appearance",
            },
        },
        defaultValue: {
            control: false,
            description: "The default value of the tabs which is used to determine which tab is selected",
            table: {
                category: "Behavior",
            },
        },
        className: {
            control: false,
            description: "Additional class name for the tabs",
            table: {
                category: "Styling",
            },
        },
        style: {
            control: false,
            description: "Additional style for the tabs",
            table: {
                category: "Styling",
            },
        },
    },
};
export default meta;
export const Default = {
    args: {
        variant: "clear",
        defaultValue: "tab1",
    },
    render: (args) => (_jsx(Card, { children: _jsxs(Tabs, { ...args, children: [_jsxs(TabsList, { children: [_jsx(TabsTrigger, { value: "tab1", text: "Paris" }), _jsx(TabsTrigger, { value: "tab2", text: "Tokyo" }), _jsx(TabsTrigger, { value: "tab3", text: "New York" }), _jsx(TabsTrigger, { value: "tab1", text: "Paris" }), _jsx(TabsTrigger, { value: "tab2", text: "Tokyo" }), _jsx(TabsTrigger, { value: "tab3", text: "New York" }), _jsx(TabsTrigger, { value: "tab1", text: "Paris" }), _jsx(TabsTrigger, { value: "tab2", text: "Tokyo" }), _jsx(TabsTrigger, { value: "tab3", text: "New York" }), _jsx(TabsTrigger, { value: "tab1", text: "Paris" }), _jsx(TabsTrigger, { value: "tab2", text: "Tokyo" }), _jsx(TabsTrigger, { value: "tab3", text: "New York" })] }), _jsxs(TabsContent, { value: "tab1", children: [_jsx(CardHeader, { title: "Paris, France", subtitle: "The City of Light", actions: [_jsx(IconButton, { variant: "tertiary", size: "small", icon: _jsx(PlaneTakeoff, {}) })] }), _jsx(Image, { src: "https://images.unsplash.com/photo-1502602898657-3e91760cbb34", alt: "Eiffel Tower in Paris", scale: "fill" })] }), _jsxs(TabsContent, { value: "tab2", children: [_jsx(CardHeader, { title: "Tokyo, Japan", subtitle: "Where Tradition Meets Future", actions: [_jsx(IconButton, { variant: "tertiary", size: "small", icon: _jsx(Ship, {}) })] }), _jsx(Image, { src: "https://images.unsplash.com/photo-1536098561742-ca998e48cbcc", alt: "Tokyo cityscape with Mount Fuji", scale: "fill" })] }), _jsxs(TabsContent, { value: "tab3", children: [_jsx(CardHeader, { title: "New York City, USA", subtitle: "The City That Never Sleeps", actions: [_jsx(IconButton, { variant: "tertiary", size: "small", icon: _jsx(BusFront, {}) })] }), _jsx(Image, { src: "https://images.unsplash.com/photo-1496442226666-8d4d0e62e6e9", alt: "New York City skyline", scale: "fill" })] })] }) })),
};
export const WithIcon = {
    args: {
        variant: "clear",
        defaultValue: "tab2",
    },
    render: (args) => (_jsxs(Tabs, { ...args, children: [_jsxs(TabsList, { children: [_jsx(TabsTrigger, { value: "tab1", text: "Venice", icon: _jsx(PlaneTakeoff, {}) }), _jsx(TabsTrigger, { value: "tab2", text: "Kyoto", icon: _jsx(Ship, {}) }), _jsx(TabsTrigger, { value: "tab3", text: "Dubai", icon: _jsx(BusFront, {}) })] }), _jsxs(TabsContent, { value: "tab1", children: [_jsx(CardHeader, { title: "Venice, Italy", subtitle: "The Floating City", actions: [_jsx(IconButton, { variant: "tertiary", size: "small", icon: _jsx(PlaneTakeoff, {}) })] }), _jsx(Image, { src: "https://images.unsplash.com/photo-1514890547357-a9ee288728e0", alt: "Venice canals and gondolas", scale: "fill" })] }), _jsxs(TabsContent, { value: "tab2", children: [_jsx(CardHeader, { title: "Kyoto, Japan", subtitle: "The Cultural Heart of Japan", actions: [_jsx(IconButton, { variant: "tertiary", size: "small", icon: _jsx(Ship, {}) })] }), _jsx(Image, { src: "https://images.unsplash.com/photo-1493976040374-85c8e12f0c0e", alt: "Traditional temple in Kyoto", scale: "fill" })] }), _jsxs(TabsContent, { value: "tab3", children: [_jsx(CardHeader, { title: "Dubai, UAE", subtitle: "City of the Future", actions: [_jsx(IconButton, { variant: "tertiary", size: "small", icon: _jsx(BusFront, {}) })] }), _jsx(Image, { src: "https://images.unsplash.com/photo-1512453979798-5ea266f8880c", alt: "Dubai skyline", scale: "fill" })] })] })),
};
