// @crayonai/react-ui/dist/components/CopilotShell/stories/Shell.stories.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { ChatProvider, useThreadListManager, useThreadManager, } from "@crayonai/react-core";
import { Composer, Container, Header, MessageLoading, Messages, ScrollArea, ThreadContainer, } from "../../CopilotShell";
// @ts-ignore
import styles from "./style.module.scss";
import logoUrl from "./thesysdev_logo.jpeg";
export default {
    title: "Copilot Shell",
    tags: ["!dev", "!autodocs"],
};
export const Default = {
    render: (args) => {
        const threadListManager = useThreadListManager({
            createThread: async () => {
                return {
                    threadId: crypto.randomUUID(),
                    title: "test",
                    createdAt: new Date(),
                    isRunning: false,
                };
            },
            fetchThreadList: async () => {
                await new Promise((resolve) => setTimeout(resolve, 1000));
                return [
                    {
                        threadId: "1",
                        title: "test",
                        createdAt: new Date(),
                        isRunning: false,
                    },
                    {
                        threadId: "2",
                        title: "test 2",
                        createdAt: new Date(),
                        isRunning: false,
                    },
                    {
                        threadId: "3",
                        title: "test 3",
                        createdAt: new Date(),
                        isRunning: false,
                    },
                ];
            },
            deleteThread: async () => { },
            updateThread: async (t) => t,
            onSwitchToNew: () => { },
            onSelectThread: () => { },
        });
        const threadManager = useThreadManager({
            threadId: threadListManager.selectedThreadId,
            loadThread: async () => {
                return [
                    {
                        id: crypto.randomUUID(),
                        role: "user",
                        type: "prompt",
                        message: "Hello",
                    },
                    {
                        id: crypto.randomUUID(),
                        role: "assistant",
                        type: "response",
                        message: [{ type: "text", text: "Hello" }],
                    },
                ];
            },
            onProcessMessage: async ({ message, threadManager, abortController }) => {
                const newMessage = Object.assign({}, message, {
                    id: crypto.randomUUID(),
                });
                threadManager.appendMessages(newMessage);
                await new Promise((resolve) => setTimeout(resolve, 1000));
                return [
                    {
                        id: crypto.randomUUID(),
                        role: "assistant",
                        type: "response",
                        message: [{ type: "text", text: "sadfasdf" }],
                    },
                ];
            },
            responseTemplates: [],
        });
        return (_jsxs("div", { className: styles.container, children: [_jsx("div", { className: styles.left }), _jsx(ChatProvider, { threadListManager: threadListManager, threadManager: threadManager, children: _jsx(Container, { logoUrl: logoUrl, agentName: "Crayon", children: _jsxs(ThreadContainer, { children: [_jsx(Header, {}), _jsx(ScrollArea, { children: _jsx(Messages, { loader: _jsx(MessageLoading, {}) }) }), _jsx(Composer, {})] }) }) })] }));
    },
};
