// @crayonai/react-ui/dist/components/DatePicker/helpers/components/DatePickerRenderer.js
import { jsx as _jsx } from "react/jsx-runtime";
import clsx from "clsx";
import { forwardRef, useRef } from "react";
import { DayPicker } from "react-day-picker";
import { useMultipleRefs } from "../../../../hooks/useMultipleRefs";
import { useDatePicker } from "../context/DatePickerContext";
import { getDayPickerStyles } from "../utils/styles";
import { MonthsDropdown, NextMonthButton, PreviousMonthButton, YearsDropdown, } from "./HelperComponents";
const DatepickerRenderer = forwardRef(({ className, style }, ref) => {
    const { selectedDate, selectedRange, mode, botType, setSelectedDate, setSelectedRange } = useDatePicker();
    const { DateSingleClasses, DateRangeClasses } = getDayPickerStyles(botType);
    const containerRef = useRef(null);
    const assignRef = useMultipleRefs(ref, containerRef);
    const commonProps = {
        captionLayout: "dropdown",
        startMonth: new Date(1900, 0),
        endMonth: new Date(2100, 11),
        components: {
            NextMonthButton,
            PreviousMonthButton,
            MonthsDropdown: (props) => (_jsx(MonthsDropdown, { ...props, container: containerRef.current })),
            YearsDropdown: (props) => _jsx(YearsDropdown, { ...props, container: containerRef.current }),
        },
    };
    if (mode === "single") {
        return (_jsx("div", { ref: assignRef, className: clsx("crayon-date-picker-renderer-single-mode", className), style: style, children: _jsx(DayPicker, { mode: "single", selected: selectedDate, onSelect: setSelectedDate, classNames: DateSingleClasses, ...commonProps }) }));
    }
    return (_jsx("div", { ref: assignRef, className: clsx("crayon-date-picker-renderer-range-mode", className), style: style, children: _jsx(DayPicker, { mode: "range", selected: selectedRange, onSelect: setSelectedRange, classNames: DateRangeClasses, ...commonProps }) }));
});
export { DatepickerRenderer };
