// @crayonai/react-ui/dist/components/Charts/AreaChartCondensed/AreaChartCondensed.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import clsx from "clsx";
import React, { useMemo, useRef, useState } from "react";
import { Area, AreaChart as RechartsAreaChart, XAxis, YAxis } from "recharts";
import { useId } from "../../../polyfills";
import { ChartContainer, ChartTooltip } from "../Charts";
import { X_AXIS_PADDING } from "../constants";
import { SideBarTooltipProvider } from "../context/SideBarTooltipContext";
import { useAutoAngleCalculation, useMaxLabelWidth, useTransformedKeys, useYAxisLabelWidth, } from "../hooks";
import { ActiveDot, cartesianGrid, CondensedXAxisTick, CustomTooltipContent, YAxisTick, } from "../shared";
import { LabelTooltipProvider } from "../shared/LabelTooltip/LabelTooltip";
import { get2dChartConfig, getDataKeys } from "../utils/dataUtils";
import { useChartPalette } from "../utils/PalletUtils";
const CHART_HEIGHT = 200;
const CHART_CONTAINER_BOTTOM_MARGIN = 10;
const AreaChartCondensedComponent = ({ data, categoryKey, theme = "ocean", customPalette, variant = "natural", tickVariant = "singleLine", grid = true, icons = {}, isAnimationActive = false, showYAxis = true, className, height = CHART_HEIGHT, width, }) => {
    const dataKeys = useMemo(() => {
        return getDataKeys(data, categoryKey);
    }, [data, categoryKey]);
    const { yAxisWidth, setLabelWidth } = useYAxisLabelWidth(data, dataKeys);
    const maxLabelWidth = useMaxLabelWidth(data, categoryKey);
    const { angle: calculatedAngle, height: xAxisHeight } = useAutoAngleCalculation(maxLabelWidth, yAxisWidth, tickVariant === "angled", showYAxis);
    const effectiveHeight = useMemo(() => {
        if (tickVariant === "angled") {
            return xAxisHeight + height;
        }
        return height;
    }, [height, xAxisHeight]);
    const transformedKeys = useTransformedKeys(dataKeys);
    const colors = useChartPalette({
        chartThemeName: theme,
        customPalette,
        themePaletteName: "areaChartPalette",
        dataLength: dataKeys.length,
    });
    const chartConfig = useMemo(() => {
        return get2dChartConfig(dataKeys, colors, transformedKeys, undefined, icons);
    }, [dataKeys, icons, colors, transformedKeys]);
    const id = useId();
    const gradientID = useMemo(() => `area-chart-condensed-gradient-${id}`, [id]);
    const chartMargin = useMemo(() => ({
        top: 10,
        right: 10,
        bottom: CHART_CONTAINER_BOTTOM_MARGIN,
        left: showYAxis ? 10 : 0,
    }), [showYAxis]);
    const containerRef = useRef(null);
    const [isSideBarTooltipOpen, setIsSideBarTooltipOpen] = useState(false);
    const [sideBarTooltipData, setSideBarTooltipData] = useState({
        title: "",
        values: [],
    });
    return (_jsx(LabelTooltipProvider, { children: _jsx(SideBarTooltipProvider, { isSideBarTooltipOpen: isSideBarTooltipOpen, setIsSideBarTooltipOpen: setIsSideBarTooltipOpen, data: sideBarTooltipData, setData: setSideBarTooltipData, children: _jsx("div", { ref: containerRef, className: clsx("crayon-area-chart-condensed", className), style: {
                    width: width ? `${width}px` : "100%",
                    height: `${effectiveHeight}px`,
                }, children: _jsx(ChartContainer, { config: chartConfig, style: { width: "100%", height: "100%" }, rechartsProps: {
                        width: "100%",
                        height: "100%",
                    }, children: _jsxs(RechartsAreaChart, { accessibilityLayer: true, data: data, margin: chartMargin, children: [grid && cartesianGrid(), _jsx(XAxis, { dataKey: categoryKey, tickLine: false, axisLine: false, textAnchor: tickVariant === "angled" ? "end" : "middle", interval: "preserveStartEnd", minTickGap: 5, height: xAxisHeight, tick: _jsx(CondensedXAxisTick, {}), angle: calculatedAngle, orientation: "bottom", padding: {
                                    left: X_AXIS_PADDING,
                                    right: X_AXIS_PADDING,
                                } }), showYAxis && (_jsx(YAxis, { tickLine: false, axisLine: false, tick: _jsx(YAxisTick, { setLabelWidth: setLabelWidth }), width: yAxisWidth })), _jsx(ChartTooltip, { content: _jsx(CustomTooltipContent, { parentRef: containerRef }), offset: 10 }), dataKeys.map((key) => {
                                const transformedKey = transformedKeys[key];
                                const color = `var(--color-${transformedKey})`;
                                return (_jsx("defs", { children: _jsxs("linearGradient", { id: `${gradientID}-${transformedKey}`, x1: "0", y1: "0", x2: "0", y2: "1", children: [_jsx("stop", { offset: "5%", stopColor: color, stopOpacity: 0.6 }), _jsx("stop", { offset: "95%", stopColor: color, stopOpacity: 0 })] }) }, `gradient-${transformedKey}`));
                            }), dataKeys.map((key) => {
                                const transformedKey = transformedKeys[key];
                                const color = `var(--color-${transformedKey})`;
                                return (_jsx(Area, { dataKey: key, type: variant, stroke: color, fill: `url(#${gradientID}-${transformedKey})`, fillOpacity: 1, stackId: "a", activeDot: _jsx(ActiveDot, {}, `active-dot-${key}-${id}`), dot: false, isAnimationActive: isAnimationActive }, `area-${key}`));
                            })] }, `area-chart-condensed-${id}`) }) }) }) }));
};
// Added React.memo for performance optimization to avoid unnecessary re-renders
export const AreaChartCondensed = React.memo(AreaChartCondensedComponent);
