// @crayonai/react-ui/dist/components/Shell/Sidebar.js
import { jsx as _jsx, Fragment as _Fragment, jsxs as _jsxs } from "react/jsx-runtime";
import clsx from "clsx";
import { PanelLeft, PanelRight } from "lucide-react";
import { useEffect } from "react";
import { useLayoutContext } from "../../context/LayoutContext";
import { IconButton } from "../IconButton";
import { useShellStore } from "./store";
export const SidebarContainer = ({ children, className, }) => {
    const { isSidebarOpen, setIsSidebarOpen } = useShellStore((state) => ({
        isSidebarOpen: state.isSidebarOpen,
        setIsSidebarOpen: state.setIsSidebarOpen,
    }));
    const { layout } = useLayoutContext() || {};
    const isMobile = layout === "mobile";
    useEffect(() => {
        if (isMobile) {
            setIsSidebarOpen(false);
        }
        else {
            setIsSidebarOpen(true);
        }
    }, [isMobile]);
    return (_jsxs(_Fragment, { children: [isMobile && (_jsx("div", { className: clsx("crayon-shell-sidebar-container__overlay", {
                    "crayon-shell-sidebar-container__overlay--collapsed": !isSidebarOpen,
                }), onClick: () => {
                    setIsSidebarOpen(false);
                } })), _jsx("div", { className: clsx("crayon-shell-sidebar-container", {
                    "crayon-shell-sidebar-container--collapsed": !isSidebarOpen,
                }, className), children: children })] }));
};
export const SidebarHeader = ({ className }) => {
    const { agentName, logoUrl, setIsSidebarOpen, isSidebarOpen } = useShellStore((state) => ({
        agentName: state.agentName,
        logoUrl: state.logoUrl,
        setIsSidebarOpen: state.setIsSidebarOpen,
        isSidebarOpen: state.isSidebarOpen,
    }));
    return (_jsxs("div", { className: clsx("crayon-shell-sidebar-header", className), children: [_jsx("img", { src: logoUrl, alt: agentName, className: "crayon-shell-sidebar-header__logo" }), _jsx("div", { className: "crayon-shell-sidebar-header__agent-name", children: agentName }), _jsx(IconButton, { icon: isSidebarOpen ? _jsx(PanelRight, { size: "1em" }) : _jsx(PanelLeft, { size: "1em" }), onClick: () => {
                    setIsSidebarOpen(!isSidebarOpen);
                }, size: "large", variant: "secondary", className: "crayon-shell-sidebar-header__toggle-button" })] }));
};
export const SidebarContent = ({ children, className, }) => {
    const { isSidebarOpen } = useShellStore((state) => ({
        isSidebarOpen: state.isSidebarOpen,
    }));
    if (!isSidebarOpen) {
        return null;
    }
    return _jsx("div", { className: clsx("crayon-shell-sidebar-content", className), children: children });
};
export const SidebarSeparator = () => {
    return _jsx("div", { className: "crayon-shell-sidebar-separator" });
};
