// @crayonai/react-ui/dist/components/Calendar/components/helperComponents.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as SelectPrimitive from "@radix-ui/react-select";
import clsx from "clsx";
import debounce from "lodash-es/debounce";
import { forwardRef, useEffect, useState } from "react";
import "react-day-picker/style.css";
import { Select, SelectItem, SelectTrigger, SelectValue } from "../../Select";
import { getMonthName, getMonthNumber } from "../utils/helperFn";
// this component is overriding the default SelectContent component to allow for a container prop we do not wish to give to the user
const SelectContent = forwardRef(({ className, children, position = "popper", viewportClassName, container, ...props }, ref) => (_jsx(SelectPrimitive.Portal, { container: container || document.body, children: _jsx(SelectPrimitive.Content, { ref: ref, className: clsx("crayon-select-content", className), position: position, ...props, children: _jsx(SelectPrimitive.Viewport, { className: clsx("crayon-select-viewport", viewportClassName), "data-position": position, children: children }) }) })));
export const MonthsDropdown = (props) => {
    const { className, disabled, onChange, options, value, key, "aria-label": ariaLabel, container, } = props;
    const [containerWidth, setContainerWidth] = useState(0);
    const [containerHeight, setContainerHeight] = useState(0);
    useEffect(() => {
        if (!container)
            return;
        const targetElement = container.children[0]?.children[0];
        if (!targetElement)
            return;
        const resizeObserver = new ResizeObserver(debounce((entries) => {
            const { width, height } = entries[0]?.contentRect || {};
            setContainerWidth(width || 0);
            setContainerHeight(height || 0);
        }, 100));
        resizeObserver.observe(targetElement);
        return () => {
            resizeObserver.disconnect();
        };
    }, [container]);
    return (_jsxs(Select, { onValueChange: (value) => onChange?.({
            target: { value: getMonthNumber(value) },
        }), value: getMonthName(Number(value)), disabled: disabled, "aria-label": ariaLabel, children: [_jsx(SelectTrigger, { className: clsx("crayon-calendar-select-trigger", className), children: _jsx(SelectValue, { placeholder: "Select a month" }) }), _jsx(SelectContent, { container: container, className: "crayon-calendar-select-content-months", sideOffset: 12, alignOffset: 0, style: {
                    maxHeight: `${containerHeight - 45}px`,
                    minWidth: `${containerWidth}px`,
                    border: "none",
                }, children: options?.map((option) => (_jsx(SelectItem, { value: getMonthName(option.value), disabled: option.disabled, children: option.label }, option.value))) })] }, key));
};
export const YearsDropdown = (props) => {
    const { className, disabled, onChange, options, value, key, "aria-label": ariaLabel, container, botType, } = props;
    const [containerWidth, setContainerWidth] = useState(0);
    const [containerHeight, setContainerHeight] = useState(0);
    useEffect(() => {
        if (!container)
            return;
        const targetElement = container.children[0]?.children[0];
        if (!targetElement)
            return;
        const resizeObserver = new ResizeObserver(debounce((entries) => {
            const { width, height } = entries[0]?.contentRect || {};
            setContainerWidth(width || 0);
            setContainerHeight(height || 0);
        }, 100));
        resizeObserver.observe(targetElement);
        return () => {
            resizeObserver.disconnect();
        };
    }, [container]);
    return (_jsxs(Select, { onValueChange: (value) => onChange?.({
            target: { value: Number(value) },
        }), value: String(value), disabled: disabled, "aria-label": ariaLabel, children: [_jsx(SelectTrigger, { className: clsx("crayon-calendar-select-trigger", className), children: _jsx(SelectValue, { placeholder: "Select a month" }) }), _jsx(SelectContent, { container: container, className: "crayon-calendar-select-content-years", viewportClassName: clsx("crayon-calendar-select-viewport", botType === "mobile" && "crayon-calendar-select-viewport-mobile"), sideOffset: 12, alignOffset: -75, style: {
                    minHeight: `${containerHeight - 45}px`,
                    maxHeight: `${containerHeight - 45}px`,
                    minWidth: `${containerWidth + 10}px`,
                    maxWidth: `${containerWidth + 10}px`,
                    border: "none",
                }, children: options?.map((option) => (_jsx(SelectItem, { value: String(option.value), disabled: option.disabled, showTick: false, className: "crayon-calendar-select-item", children: option.label }, option.value))) })] }, key));
};
