// @crayonai/react-ui/dist/components/Charts/ScatterChart/ScatterChart.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import clsx from "clsx";
import { useEffect, useMemo, useRef, useState } from "react";
import { Cell, ScatterChart as RechartsScatterChart, Scatter, XAxis, YAxis } from "recharts";
import { useId } from "../../../polyfills";
import { ChartContainer, ChartTooltip } from "../Charts";
import { SideBarTooltipProvider } from "../context/SideBarTooltipContext";
import { useYAxisLabelWidth } from "../hooks";
import { CustomTooltipContent, DefaultLegend, gridCartesianGrid, SideBarTooltip, YAxisTick, } from "../shared";
import { get2dChartConfig, getLegendItems } from "../utils/dataUtils";
import { useChartPalette } from "../utils/PalletUtils";
import { numberTickFormatter } from "../utils/styleUtils";
import ScatterDot from "./components/ScatterDot";
import { calculateScatterDomain, getScatterDatasets, transformScatterData, } from "./utils/ScatterChartUtils";
const DEFAULT_CHART_HEIGHT = 296;
const DEFAULT_MARGIN = 20;
const X_AXIS_HEIGHT = 40;
const Y_AXIS_OFFSET = 10;
export const ScatterChart = ({ data, xAxisDataKey = "x", yAxisDataKey = "y", theme = "ocean", customPalette, grid = true, xAxisLabel, yAxisLabel, legend = true, isAnimationActive = false, className, height, width, shape = "circle", }) => {
    const datasets = useMemo(() => {
        return getScatterDatasets(data);
    }, [data]);
    const colors = useChartPalette({
        chartThemeName: theme,
        customPalette,
        themePaletteName: "lineChartPalette",
        dataLength: datasets.length,
    });
    const transformedData = useMemo(() => {
        if (!data || !Array.isArray(data)) {
            return [];
        }
        return transformScatterData(data, datasets, colors);
    }, [data, datasets, colors]);
    const { yAxisWidth, setLabelWidth } = useYAxisLabelWidth(transformedData, [yAxisDataKey]);
    const chartConfig = useMemo(() => {
        return get2dChartConfig(datasets, colors, datasets.reduce((acc, key) => ({ ...acc, [key]: key }), {}), undefined);
    }, [datasets, colors]);
    const chartWrapperRef = useRef(null);
    const legendContainerRef = useRef(null);
    const xAxisContainerRef = useRef(null);
    const [containerWidth, setContainerWidth] = useState(0);
    const [isSideBarTooltipOpen, setIsSideBarTooltipOpen] = useState(false);
    const [isLegendExpanded, setIsLegendExpanded] = useState(false);
    const [sideBarTooltipData, setSideBarTooltipData] = useState({
        title: "",
        values: [],
    });
    const isFixedNumericHeight = useMemo(() => {
        if (typeof height === "number")
            return true;
        if (typeof height === "string" && height.endsWith("px"))
            return true;
        return false;
    }, [height]);
    const chartHeight = useMemo(() => {
        if (!chartWrapperRef.current) {
            return 0;
        }
        const legendHeight = legendContainerRef.current?.offsetHeight ?? 0;
        const xAxisHeight = xAxisContainerRef.current?.offsetHeight ?? 0;
        if (typeof height === "number") {
            return height - legendHeight - xAxisHeight;
        }
        return chartWrapperRef.current.offsetHeight - legendHeight - xAxisHeight;
    }, [containerWidth, height]);
    // Calculate domains for x and y axes
    const xDomain = useMemo(() => {
        return calculateScatterDomain(data, xAxisDataKey);
    }, [data, xAxisDataKey]);
    const yDomain = useMemo(() => {
        return calculateScatterDomain(data, yAxisDataKey);
    }, [data, yAxisDataKey]);
    const renderDotShape = useMemo(() => {
        return (props) => {
            return _jsx(ScatterDot, { ...props, variant: shape });
        };
    }, [shape]);
    useEffect(() => {
        const chartElement = chartWrapperRef.current;
        if (!chartElement) {
            return;
        }
        const resizeObserver = new ResizeObserver((entries) => {
            for (const entry of entries) {
                if (entry.target === chartElement) {
                    setContainerWidth(entry.contentRect.width);
                }
            }
        });
        resizeObserver.observe(chartElement);
        return () => {
            resizeObserver.disconnect();
        };
    }, []);
    useEffect(() => {
        setIsSideBarTooltipOpen(false);
        setIsLegendExpanded(false);
    }, [datasets]);
    const legendItems = useMemo(() => {
        return getLegendItems(datasets, colors);
    }, [datasets, colors]);
    const id = useId();
    const xAxis = useMemo(() => {
        return (_jsx("div", { className: "crayon-scatter-chart-x-axis-container-outer", ref: xAxisContainerRef, children: _jsx("div", { className: "crayon-scatter-chart-x-axis-container", style: containerWidth
                    ? {
                        width: containerWidth - yAxisWidth - Y_AXIS_OFFSET,
                        marginLeft: yAxisWidth + Y_AXIS_OFFSET,
                    }
                    : {
                        width: "90%",
                    }, children: _jsx(ChartContainer, { config: chartConfig, style: { width: "100%", height: X_AXIS_HEIGHT }, rechartsProps: {
                        height: X_AXIS_HEIGHT,
                    }, children: _jsxs(RechartsScatterChart, { data: transformedData, margin: {
                            top: 10,
                            bottom: 0,
                            left: DEFAULT_MARGIN - 3,
                            right: 2,
                        }, children: [_jsx(XAxis, { type: "number", height: X_AXIS_HEIGHT, name: xAxisLabel, tickLine: false, axisLine: false, tickFormatter: numberTickFormatter, tick: { fontSize: 12 }, domain: xDomain, dataKey: xAxisDataKey }), _jsx(Scatter, { data: transformedData, fill: "transparent", isAnimationActive: isAnimationActive, shape: "circle" })] }, `x-axis-scatter-chart-${id}`) }) }) }));
    }, [
        chartConfig,
        transformedData,
        id,
        yAxisWidth,
        xDomain,
        xAxisDataKey,
        isAnimationActive,
        containerWidth,
        xAxisLabel,
    ]);
    return (_jsx(SideBarTooltipProvider, { isSideBarTooltipOpen: isSideBarTooltipOpen, setIsSideBarTooltipOpen: setIsSideBarTooltipOpen, data: sideBarTooltipData, setData: setSideBarTooltipData, children: _jsxs("div", { className: clsx("crayon-scatter-chart-container", className), style: {
                width: typeof width === "number" ? `${width}px` : width || "100%",
                height: isFixedNumericHeight ? "auto" : height || "100%",
            }, ref: chartWrapperRef, children: [_jsxs("div", { className: "crayon-scatter-chart-container-inner", children: [_jsx("div", { className: "crayon-scatter-chart-main-container", style: {
                                width: "100%",
                                height: typeof height === "number" ? `${height}px` : height || DEFAULT_CHART_HEIGHT,
                            }, children: _jsx(ChartContainer, { config: chartConfig, style: {
                                    width: "100%",
                                    height: "100%",
                                    aspectRatio: 0,
                                }, rechartsProps: {
                                    width: "100%",
                                    height: "100%",
                                }, children: _jsxs(RechartsScatterChart, { margin: {
                                        bottom: 10,
                                    }, children: [grid && gridCartesianGrid({ horizontal: true, vertical: true }), _jsx(XAxis, { type: "number", height: X_AXIS_HEIGHT, name: xAxisLabel, tickLine: false, axisLine: false, tickFormatter: numberTickFormatter, tick: { fontSize: 12 }, domain: xDomain, dataKey: xAxisDataKey, hide: true }), _jsx(YAxis, { type: "number", dataKey: yAxisDataKey, name: yAxisLabel, domain: yDomain, tickLine: false, axisLine: false, tick: _jsx(YAxisTick, { setLabelWidth: setLabelWidth }), tickFormatter: numberTickFormatter }), _jsx(ChartTooltip, { content: _jsx(CustomTooltipContent, { parentRef: chartWrapperRef, hideIndicator: true }), offset: 15 }), _jsx(Scatter, { data: transformedData, shape: renderDotShape, isAnimationActive: isAnimationActive, children: transformedData.map((entry, index) => (_jsx(Cell, { fill: entry["color"] }, `cell-${index}`))) }, `scatter-${id}`)] }, `scatter-chart-${id}`) }) }), isSideBarTooltipOpen && chartHeight > 0 && _jsx(SideBarTooltip, { height: chartHeight })] }), xAxis, _jsx("div", { className: "crayon-scatter-chart-legend-container", ref: legendContainerRef, children: legend && (_jsx(DefaultLegend, { items: legendItems, yAxisLabel: yAxisLabel, xAxisLabel: xAxisLabel, containerWidth: containerWidth, isExpanded: isLegendExpanded, setIsExpanded: setIsLegendExpanded })) })] }) }));
};
