// @crayonai/react-ui/dist/components/MarkDownRenderer/MarkDownRenderer.js
import { jsx as _jsx } from "react/jsx-runtime";
import clsx from "clsx";
import { memo } from "react";
import ReactMarkdown from "react-markdown";
import { oneLight, vscDarkPlus } from "react-syntax-highlighter/dist/esm/styles/prism";
import { CodeBlock } from "../CodeBlock";
import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from "../Table";
import { useTheme } from "../ThemeProvider";
const variantStyles = {
    clear: "",
    card: "crayon-markdown-renderer-card",
    sunk: "crayon-markdown-renderer-card-sunk",
};
export const MarkDownRenderer = memo((props) => {
    const { mode } = useTheme();
    const theme = mode === "dark" ? vscDarkPlus : oneLight;
    const components = {
        code({ className, children, ...props }) {
            const match = /language-(\w+)/.exec(className || "");
            if (match || (!className && String(children).includes("\n"))) {
                const language = match?.[1] ?? "text";
                const codeString = String(children).trim();
                return _jsx(CodeBlock, { language: language, codeString: codeString, theme: theme });
            }
            return (_jsx("code", { className: clsx("crayon-markdown-renderer-code", className), ...props, children: children }));
        },
        a({ href, children, ...props }) {
            return (_jsx("a", { href: href, target: "_blank", rel: "noopener noreferrer", className: "crayon-markdown-renderer-link", ...props, children: children }));
        },
        table: Table,
        thead: TableHeader,
        th: TableHead,
        tbody: TableBody,
        tr: TableRow,
        td: TableCell,
    };
    const markdownProps = {
        ...props.options,
        components: { ...components, ...props.options?.components },
    };
    return (_jsx("div", { className: clsx(props["variant"] && variantStyles[props["variant"]], "crayon-markdown-renderer", props.className), children: _jsx(ReactMarkdown, { ...markdownProps, children: props.textMarkdown }) }));
});
