// @crayonai/react-core/dist/internal/useThreadManagerStore.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useThreadManagerStore = void 0;
const react_1 = require("react");
const zustand_1 = require("zustand");
// this is to create zustand store on top of input thread manager,
// it enables us to expose dev friendly api using context and not give up on optimisations
const useThreadManagerStore = (inputThreadManager) => {
    const inputThreadManagerRef = (0, react_1.useRef)(inputThreadManager);
    inputThreadManagerRef.current = inputThreadManager;
    const [threadManagerStore] = (0, react_1.useState)(() => (0, zustand_1.create)(() => ({
        isLoadingMessages: inputThreadManager.isLoadingMessages,
        isRunning: inputThreadManager.isRunning,
        messages: inputThreadManager.messages,
        error: inputThreadManager.error,
        responseTemplates: inputThreadManagerRef.current.responseTemplates,
        onCancel: () => inputThreadManagerRef.current.onCancel?.(),
        processMessage: (...props) => inputThreadManagerRef.current.processMessage(...props),
        updateMessage: (...props) => inputThreadManagerRef.current.updateMessage(...props),
        appendMessages: (...props) => inputThreadManagerRef.current.appendMessages(...props),
        setMessages: (...props) => inputThreadManagerRef.current.setMessages(...props),
        deleteMessage: (...props) => inputThreadManagerRef.current.deleteMessage(...props),
    })));
    (0, react_1.useEffect)(() => {
        threadManagerStore.setState({
            isRunning: inputThreadManager.isRunning,
            messages: inputThreadManager.messages,
            error: inputThreadManager.error,
            isLoadingMessages: inputThreadManager.isLoadingMessages,
        });
    }, [
        inputThreadManager.isRunning,
        inputThreadManager.messages,
        inputThreadManager.error,
        inputThreadManager.isLoadingMessages,
    ]);
    if (inputThreadManagerRef.current.responseTemplates !== inputThreadManager.responseTemplates) {
        throw new Error("memoize response templates");
    }
    return threadManagerStore;
};
exports.useThreadManagerStore = useThreadManagerStore;
