// @crayonai/react-core/dist/internal/useThreadListManagerStore.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useThreadListManagerStore = void 0;
const react_1 = require("react");
const zustand_1 = require("zustand");
// this is to create zustand store on top of input thread list manager,
// it enables us to expose dev friendly api using context and not give up on optimisations
const useThreadListManagerStore = (inputThreadListManager) => {
    const inputThreadListManagerRef = (0, react_1.useRef)(inputThreadListManager);
    inputThreadListManagerRef.current = inputThreadListManager;
    const [threadListManagerStore] = (0, react_1.useState)(() => (0, zustand_1.createStore)(() => {
        return {
            threads: inputThreadListManagerRef.current.threads,
            isLoading: inputThreadListManagerRef.current.isLoading,
            error: inputThreadListManagerRef.current.error,
            selectedThreadId: inputThreadListManagerRef.current.selectedThreadId,
            shouldResetThreadState: inputThreadListManagerRef.current.shouldResetThreadState,
            switchToNewThread: () => inputThreadListManagerRef.current.switchToNewThread(),
            load: () => inputThreadListManagerRef.current.load(),
            updateThread: (...props) => inputThreadListManagerRef.current.updateThread(...props),
            deleteThread: (...props) => inputThreadListManagerRef.current.deleteThread(...props),
            selectThread: (...props) => inputThreadListManagerRef.current.selectThread(...props),
            createThread: (...props) => inputThreadListManagerRef.current.createThread(...props),
        };
    }));
    (0, react_1.useEffect)(() => {
        threadListManagerStore.setState({
            threads: inputThreadListManager.threads,
            isLoading: inputThreadListManager.isLoading,
            error: inputThreadListManager.error,
            selectedThreadId: inputThreadListManager.selectedThreadId,
            shouldResetThreadState: inputThreadListManager.shouldResetThreadState,
        });
        // no dependency array, since zustand will handle rerendering automatically
    });
    return threadListManagerStore;
};
exports.useThreadListManagerStore = useThreadListManagerStore;
