// @crayonai/react-ui/dist/components/ListBlock/stories/ListBlock.stories.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { ChevronRight, User } from "lucide-react";
import { ListItem } from "../../ListItem";
import { ListBlock } from "../ListBlock";
const meta = {
    title: "Components/ListBlock",
    component: ListBlock,
    subcomponents: {
        ListItem: ListItem,
    },
    parameters: {
        layout: "centered",
        docs: {
            description: {
                component: "```tsx\nimport { ListBlock, ListItem } from '@crayon-ui/react-ui';\n```",
            },
        },
    },
    decorators: [
        (Story) => (_jsx("div", { style: { width: "400px" }, children: _jsx(Story, {}) })),
    ],
    argTypes: {
        children: {
            control: false,
            description: "Accepts ListItem components as children",
            table: {
                category: "Content",
                type: { summary: "ListItemProps" },
            },
        },
        className: {
            control: false,
            description: "Additional CSS class name for custom styling",
            table: {
                category: "Appearance",
            },
        },
        style: {
            control: false,
            description: "Additional CSS style for custom styling",
            table: {
                category: "Appearance",
            },
        },
    },
    tags: ["!dev", "autodocs"],
};
export default meta;
export const ListBlockWithItems = {
    render: (args) => (_jsxs(ListBlock, { children: [_jsx(ListItem, { decorativeIcon: _jsx(User, { size: 16 }), title: "John Doe", subtitle: "Software Engineer", actionIcon: _jsx(ChevronRight, { size: 16 }) }), _jsx(ListItem, { decorativeIcon: _jsx(User, { size: 16 }), title: "Jane Smith", subtitle: "Product Designer", actionIcon: _jsx(ChevronRight, { size: 16 }) }), _jsx(ListItem, { decorativeIcon: _jsx(User, { size: 16 }), title: "Ankit Das", subtitle: "Software Developer", actionIcon: _jsx(ChevronRight, { size: 16 }) })] })),
};
