// @crayonai/react-ui/dist/components/ListItem/stories/ListItem.stories.js
import { jsx as _jsx } from "react/jsx-runtime";
import { ChevronRight, User } from "lucide-react";
import { ListItem } from "../ListItem";
const meta = {
    title: "Components/ListItem",
    component: ListItem,
    tags: ["!dev", "!autodocs"],
    parameters: {
        layout: "centered",
        docs: {
            description: {
                component: "```tsx\nimport { ListItem } from '@crayon-ui/react-ui';\n```",
            },
        },
    },
    argTypes: {
        title: {
            control: "text",
            description: "The main title text of the list item",
            table: {
                category: "Content",
                type: { summary: "ReactNode" },
            },
        },
        subtitle: {
            control: "text",
            description: "The secondary text of the list item",
            table: {
                category: "Content",
                type: { summary: "ReactNode" },
            },
        },
        decorativeIcon: {
            control: false,
            description: "The decorative icon or image to display on the left side of the list item",
            table: {
                category: "Content",
                type: { summary: "ReactNode" },
            },
        },
        actionIcon: {
            control: false,
            description: "The action icon to display on the right side of the list item",
            table: {
                category: "Content",
                type: { summary: "ReactNode" },
            },
        },
        onClick: {
            control: false,
            description: "The function to call when the list item is clicked",
            table: {
                category: "Behavior",
                type: { summary: "function" },
            },
        },
        className: {
            control: false,
            description: "Additional CSS class name for custom styling",
            table: {
                category: "Appearance",
            },
        },
        style: {
            control: false,
            description: "Additional CSS style for custom styling",
            table: {
                category: "Appearance",
            },
        },
    },
};
export default meta;
export const ListItemStory = {
    args: {
        decorativeIcon: _jsx(User, { size: 16 }),
        title: "List Item Title",
        subtitle: "Subtitle text goes here",
        actionIcon: _jsx(ChevronRight, { size: 16 }),
    },
};
