// @crayonai/react-ui/dist/components/Tabs/Tabs.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as TabsPrimitive from "@radix-ui/react-tabs";
import clsx from "clsx";
import { ChevronLeft, ChevronRight } from "lucide-react";
import { forwardRef, useEffect, useRef, useState } from "react";
import { IconButton } from "../IconButton";
const tabsVariants = {
    clear: "crayon-tabs-clear",
    card: "crayon-tabs-card",
    sunk: "crayon-tabs-sunk",
};
export const Tabs = forwardRef(({ className, style, variant = "clear", ...props }, ref) => (_jsx(TabsPrimitive.Root, { ref: ref, className: clsx("crayon-tabs", tabsVariants[variant], className), style: style, ...props })));
export const TabsList = forwardRef(({ className, style, ...props }, ref) => {
    const listRef = useRef(null);
    const [showLeftButton, setShowLeftButton] = useState(false);
    const [showRightButton, setShowRightButton] = useState(false);
    // Check if scrolling is needed
    useEffect(() => {
        const checkScroll = () => {
            if (listRef.current) {
                const { scrollLeft, scrollWidth, clientWidth } = listRef.current;
                setShowLeftButton(scrollLeft > 0);
                setShowRightButton(scrollLeft < scrollWidth - clientWidth - 1); // -1 for rounding errors
            }
        };
        // Initial check
        checkScroll();
        // Add event listener for scroll
        const currentRef = listRef.current;
        if (currentRef) {
            currentRef.addEventListener("scroll", checkScroll);
            // Also add resize observer to handle responsive changes
            const resizeObserver = new ResizeObserver(checkScroll);
            resizeObserver.observe(currentRef);
            return () => {
                currentRef.removeEventListener("scroll", checkScroll);
                resizeObserver.disconnect();
            };
        }
        return () => { };
    }, []);
    // Scroll functions
    const scrollLeft = () => {
        if (listRef.current) {
            // Scroll one tab to the left
            listRef.current.scrollBy({ left: -120, behavior: "smooth" });
        }
    };
    const scrollRight = () => {
        if (listRef.current) {
            // Scroll one tab to the right
            listRef.current.scrollBy({ left: 120, behavior: "smooth" });
        }
    };
    return (_jsxs("div", { className: "crayon-tabs-list-container", children: [showLeftButton && (_jsx("div", { className: "crayon-tabs-scroll-button-container-left", children: _jsx(IconButton, { className: "crayon-tabs-scroll-button crayon-tabs-scroll-left", onClick: scrollLeft, "aria-label": "Scroll tabs left", icon: _jsx(ChevronLeft, {}), variant: "secondary", size: "small" }) })), _jsx(TabsPrimitive.List, { ref: (node) => {
                    // Forward the ref to the original ref prop if provided
                    if (typeof ref === "function") {
                        ref(node);
                    }
                    else if (ref) {
                        ref.current = node;
                    }
                    // Also store it in our local ref
                    listRef.current = node;
                }, className: clsx("crayon-tabs-list", className), style: style, ...props }), showRightButton && (_jsx("div", { className: "crayon-tabs-scroll-button-container-right", children: _jsx(IconButton, { className: "crayon-tabs-scroll-button crayon-tabs-scroll-right", onClick: scrollRight, "aria-label": "Scroll tabs right", icon: _jsx(ChevronRight, {}), variant: "secondary", size: "small" }) }))] }));
});
export const TabsTrigger = forwardRef(({ className, style, icon, text, value, ...props }, ref) => (_jsxs(TabsPrimitive.Trigger, { ref: ref, className: clsx("crayon-tabs-trigger", className), style: style, value: value, ...props, children: [icon && _jsx("span", { className: "crayon-tabs-trigger-icon", children: icon }), text && _jsx("span", { className: "crayon-tabs-trigger-text", children: text })] })));
export const TabsContent = forwardRef(({ className, style, children, ...props }, ref) => (_jsx(TabsPrimitive.Content, { ref: ref, className: clsx("crayon-tabs-content", className), style: style, ...props, children: _jsx("div", { className: "crayon-tabs-content-inner", children: children }) })));
